### ISSUE TITLE
CodePush retirement on March 31 2025

### ISSUE THREAD

**Issue by mgscreativa** [deggertsen/react-native-code-push-expo-plugin#8]
HI @GSTJ and @deggertsen I've send an email to Microsoft support about what they will do about CodePush, ¿do you know about any other alternative without using Expo updates?

**Comment by GSTJ** [comment]
Sad news, indeed.

My recommendation is to self-host an Expo Server and use Expo Updates.

Codepush also provides self-hosting options during their departure, but these will be likely unmaintained, so they're not the best option.

**Comment by GSTJ** [comment]
https://docs.expo.dev/eas-update/custom-updates-server/

**Comment by mgscreativa** [comment]
Yup, I know about the open specification for Expo updates, but, ¿do you know of a PHP implementation?

**Comment by deggertsen** [comment]
I'm waiting for Microsoft to follow up with me on the codepush codebase. I'm not sure what they are planning, but they told me "Hello David, we will provide access to the CodePush repository, once the process is complete I will follow up."

So I'm waiting to hear back from them.

**Comment by mgscreativa** [comment]
Ok, lets use this issue to follow up this matter

**Comment by deggertsen** [comment]
I followed up with them again and they got right back to me saying "We still don't have an ETA, but I will follow up soon, thanks!"

**Comment by mgscreativa** [comment]
Hi guys! seems CP server is already available https://github.com/microsoft/code-push-server/blob/main/api/README.md

I guess if you had the opportunity to set it up, I have an Ubuntu server but din't try to set up an instance yet

**Comment by deggertsen** [comment]
One thing to note with codepush is that the new react native architecture is not supported according to this thread: https://github.com/expo/config-plugins/pull/204#issuecomment-2472915034

It might be worth switching to your own EAS server rather than running your own CP server.

**Comment by mgscreativa** [comment]
Yup, that seems to be the right path.

Do you know a PHP implementation of the EAS server?

**Comment by mgscreativa** [comment]
Just to mention this guy has an implementation with Docker container included. Didn't tested it yet, but surely will.

https://github.com/umbertoghio/self-hosted-expo-updates-server

**Comment by mgscreativa** [comment]
Another tool to play around https://github.com/xavia-io/xavia-ota

I was unable to get either project to work, do you guys @deggertsen @GSTJ have implemented self hosted OTA? how? do you plan to publish your server implementation?

Thanks!
